### ISSUE TITLE
Function to resolve static styles gets incomplete site variables

### ISSUE THREAD

**Issue by vadimvar** [microsoft/fluent-ui-react#526]
### Steps
          <UIProvider
            theme={{
              staticStyles: [
                     resolveThemedStyles,
                     …...
              ]
            }}
          >

function resolveThemedStyles(siteVariables) {
    // Site variables only contains font...
    // Where are all theme variables?
}

### Expected Result
Expect the siteVariable parameter to have correct site variables

### Actual Result
The actual result that happened 💣

### Version
x.y.z

### Testcase
[Fork, update, and replace this pen to show the bug]:
https://codesandbox.io/s/2l3n74j9y

**Comment by mnajdova** [comment]
### TL;DR
@vadimvar, please, suggest the use case so that we will be able to better address the issue. While some motivation is provided in the description, there is not enough context to be absolutely sure what this use case is about.

------------

Could you, please, suggest the use case of this? The `staticStyles` are something that is written in the body of the page, so they cannot be used for just part of the tree of the application. That means, the `staticStyles` should only be written in the root Provider, where the `siteVariables` are defined - that way you will have access to them. At this moment we are not applying to the staticStyles the merged `siteVariables` in the `Provider`, we are just using the `siteVariables` defined in the same `Provider`. 

Note, in the future, we may not even support static styles, so if there is some use case that we are missing about using them in some nested `Provider`,  please give us some example usage.

**Comment by vadimvar** [comment]
I deal with a component which is hard coding styles and the only workaround would be to use static styles. While the other team is refactoring it, we still need to unblock the scenario. This in fact could be a quite common case for teams transitioning to N* principles.
My static styles are big and I don't want to mix them with the code where root provider is defined
Lookin on static styles I see it supports conditional styling 
`export type StaticStyle = StaticStyleRenderable | StaticStyleFunction`
`export type StaticStyleFunction = (siteVariables?: SiteVariablesPrepared) => StaticStyleObject`
**Site variables is a part of signature the same as for regular styles except it is not filled.** 
This is confusing and I have explicitly reverence teams.team to resolve it.  The other concern is on how all that will work for theme other than team. Let's say in dark mode.
